core/payload/deploy_code.go

```diff
@@ -253,3 +261,45 @@ func CreateDeployCode(code []byte,
 	}
 	return contract, nil
 }
+
+type NativeDeployCode struct {
+	BaseContractAddress common.Address
+	InitParam           []byte // optional
+}
+
+func (ndc *NativeDeployCode) Serialization(sink *common.ZeroCopySink) {
+	sink.WriteAddress(ndc.BaseContractAddress)
+	if len(ndc.InitParam) > 0 {
+		sink.WriteVarBytes(ndc.InitParam)
+	}
```

Comment:
conditionally serialize struct field need to serialize a bool value first. so I suggest using `sink.WriteVarBytes(ndc.InitParam)` directly.

Reply:
OK